int32 dilema_id
bool is_receiver
bool is_safe